mteb/abstasks/AbsTaskClusteringFast.py

```diff
@@ -0,0 +1,189 @@
+    """Abstract class for Clustering tasks.
+
+    This class embeds the corpus sentences then samples N samples from the corpus and clusters them.
+    he similarity then is calculated using the V-measure metric, which is invariant to the permutation of the labels.
+    This approach is then repeated K times.
+
+    self.load_data() must generate a huggingface dataset with a split matching self.metadata_dict["eval_splits"], and assign it to self.dataset.
+    It must contain the following columns:
+        sentences: list of str
+        labels: list of str
+    """
+
+    max_documents_to_embed = 16_384
+    max_documents_per_cluster = 2048
+    n_clusters = 10
+    k_mean_batch_size = 512
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def calculate_main_score(self, scores: list[dict[str, float]]) -> float:
+        return float(np.mean([score[self.metadata.main_score] for score in scores]))
+
+    def evaluate(
+        self, model, split="test", **kwargs
+    ) -> (
+        dict[HFLang, dict[str, ClusteringScores | float]]
+        | dict[str, ClusteringScores | float]
+    ):
+        self.dataset: MultilingualDataset | HFDataset
+
+        if not self.data_loaded:
+            self.load_data()
+
+        if self.is_multilingual:
+            multilingual_ds: MultilingualDataset = self.dataset  # type: ignore
+
+            multilingual_scores: dict[HFLang, dict[str, ClusteringScores | float]] = {}
+            for lang in self.dataset:
+                logger.info(
+                    f"\nTask: {self.metadata.name}, split: {split}, language: {lang}. Running..."
+                )
+                multilingual_scores[lang] = self._evaluate_monolingual(
+                    model, multilingual_ds[lang], split, **kwargs
+                )
+                return multilingual_scores
+        logger.info(f"\nTask: {self.metadata.name}, split: {split}. Running...")
+
+        ds: HFDataset = self.dataset  # type: ignore
+        scores = self._evaluate_monolingual(model, ds, split, **kwargs)
+        return scores
+
+    def _evaluate_monolingual(
+        self, model, dataset: HFDataset, split: str = "test", **kwargs: Any
+    ) -> dict[str, float | ClusteringScores]:
+        sentences = dataset[split]["sentences"][: self.max_documents_to_embed]
+        labels = dataset[split]["labels"][: self.max_documents_to_embed]
+
+        logger.info(f"Encoding {len(labels)} sentences...")
+
+        embeddings = model.encode(sentences)  # , prompt_name=self.metadata.name)
+
+        scores = evaluate_clustering_bootstrapped(
+            embeddings,
+            labels,
+            n_clusters=self.n_clusters,
+            cluster_size=self.max_documents_per_cluster,
+            kmean_batch_size=self.k_mean_batch_size,
+            rng_state=random.Random(self.seed),
+        )
+
+        return {"main_score": self.calculate_main_score(scores), "scores": scores}
+
+
+@dataclass
+class MTEBScores:
```

Comment:
@MartinBernstorff I would love your opinion on this result object. Notably, the format of the scores seems too complex. I would love to discuss it with you.

Reply:
1. Seems like `main_score` is a subset of `scores`. Maybe we could drop it?
2. Are we missing `dataset_revision`?

Reply:
Yeah, the main score could be dropped. Will add in dataset revision

Reply:
I have removed MTEBScore - we can take it in another PR